### DISCUSSION TITLE
rAudio 20230630: GUI volume always returns to 100%

### DISCUSSION THREAD

**Discussion by princedurail** [rern/rAudio#1395] [Fixed Issues]
Hi Rern,
I use a RPI 3B+ with Hifiberry Amp and Rotary encoder.
I made the update with the 20230630 and I again have the same problem as with the version 20221117 post #1109
The volume returns to 100% systematically when I use 'mixer device' with my Hifiberry Amp card.
This does not happen with 'MPD software' yes if I disconnect my Hifiberry Amp card and use the HDMI/audio output.

amixer scontents
Simple mixer control 'Master',0
  Capabilities: volume volume-joined
  Playback channels: Mono
  Capture channels: Mono
  Limits: 0 - 248
  Mono: 81 [33%]
Simple mixer control 'Channels',0
  Capabilities: volume
  Playback channels: Front Left - Front Right
  Capture channels: Front Left - Front Right
  Limits: 0 - 248
  Front Left: 83 [33%]
  Front Right: 83 [33%]

Little video that explains the problem much better than me.
https://github.com/rern/rAudio/assets/28156398/ba62527c-0572-478e-85a6-183d84ec2d5e

thanks, Didier

**Comment by kan-ibal** [comment]
It looks like a similar issue that I've got. In my setup volume returns to 100% after a restart, instead of staying at the last value. This is an ALSA issue that is not restoring sound correctly for some of audio cards.
Check if it is the case using:
`alsactl restore`
If you see an error like 
`alsa-lib main.c:1560:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -2`
and volume returns to 100% you have a culprit.

**Comment by princedurail** [comment]
hi kan-ibal
The result of alsact1 restore is identical to yours.
  alsa-lib main.c:1560:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -2
Do you have a solution to solve this problem?

**Comment by kan-ibal** [comment]
I do not use a Rotary Encoder and it only bothered me during reboots when rAudio has started with 100% volume instead of a last set value before shutdown.
I haven't found a solution but a workaround only that I set a volume before playback starts using:
`amixer -c 1 -Mq sset "PCM" 15%`
That workaround fixes my issue and I did not dig any further.

**Comment by princedurail** [comment]
I worked around the problem of returning the volume to 100% by using 'MPD software' instead of 'DEVICE mixer', it works but forces me to use two volume settings, the Rotary encoder + that of the GUI interface, if one of the two is too low no sound.
Rern had made an update on version 20221117 which had solved the problem, I will wait for the next update.
thank you for your feedback.

**Comment by kan-ibal** [comment]
There is a problem if sound card doesn't support UCM. The solution for this issue is to append
`--no-ucm` to `alsactl` command:

`alsactl --no-ucm nrestore`

**Comment by 8e110** [comment]
Pardon my ignorance but where do I find the alsactl command?

**Reply by muddiver** [reply]
Hi there, i have the same Problem with an IQAudio Digi-AMP. This annoying return to 100% volume drives me crazy, especially for your first steps with a new system.. Can somebody please explain how i can fix this problem? I have set up MoOde and Volumio before, login per ssh is no problem, but what file should i modify? Thanks to anyone who answers! geets, Thomas

**Reply by muddiver** [reply]
Simply copying "alsactl --no-ucm nrestore" to the console when active per ssh on the pi isn´t working.. i do not get an errormesage, but my volume is still changing to 100%. What am i doing wrong? Please help me, rAudio seems to be a good OS, i like the usability with touchscreens, bot the changing Volume makes the OS unusable for me..

**Reply by rern** [reply]
With update 20231210, try: 
Settings > System > Environment > `Volume on Boot`

**Reply by muddiver** [reply]
Hello rern,
thank you for your efforts, i installed Update 20231210, unfortunely it didn't solve my issue.. It still returns to 100%. The Video in Post#1 shows exactly what happens.. every time i leave the "Player" to library or playlist, when i come back, the volume jumps to 100%.  The following applies 100% to me.. 
command: "alsactl restore"
Output:
alsa-lib main.c:1560:(snd_use_case_mgr_open) error: failed to import hw:0 use case configuration -2

Thanks for your efforts again! Greets, Thomas

**Reply by muddiver** [reply]
As an additional info, i have this amplifier: RPI HiFi AMP HAT TAS5713 Amplifier Audio Module 25W.
Normally i use the "hifiberry Amp or Amp+", which works fine in rAudio too. The issue is not only after a boot, it turns 100% Volume every time when i return to the "Player" task.... To be honest, it would not be the biggest problem to replace the amplifier hat, but i don't know which one would work.. Greets, Thomas

**Reply by muddiver** [reply]
Hi folks!
my son had the wish to build an INternetradio for himself, so hi bought a Raspberry Pi 4, a IQAudio Digiamp+.
With this configuration there is NO PROBLEM!! It seem's the Amp Hat IQAudio compatible with the Hifiberry Amp+ is the Problem.
So, i think i now have a present for myseln for christmas ;-)

Feliz navidad to everyone!

**Comment by kan-ibal** [comment]
It looks like this issue is fixed in latest alsa update fixes this issue.  `alsactl restore` works again as it suppose to.
